packages/amplify-category-geo/src/provider-utils/awscloudformation/utils/resourceUtils.ts

```diff
@@ -0,0 +1,55 @@
+import { JSONUtilities, pathManager, $TSAny, $TSContext, $TSObject } from 'amplify-cli-core';
+import { category } from '../../../constants';
+import path from 'path';
+import { MapParameters } from './mapParams';
+import _ from 'lodash';
+
+
+// Merges other with existing in a non-destructive way.
+// Specifically, scalar values will not be overwritten
+// Objects will have field added but not removed or modified
+// Arrays will be appended to, duplicates are removed
+export function merge(existing: Partial<MapParameters>, other?: Partial<MapParameters>): Partial<MapParameters> {
```

Comment:
looks like this is the same as the merge used in the functions category. We should pull this into a common location rather than duplicating. Perhaps into `amplify-cli-core`

Reply:
changed this to a generic function. Moving it to core might need some changes to the functions category and some additional testing to make sure we do not break anything there. Can I take a TODO item and do this outside this PR?